src/main/java/marquez/api/resources/JobResource.java

```diff
@@ -77,6 +78,11 @@ public Response create(
     if (!namespaceService.exists(namespaceName)) {
       return Response.status(Response.Status.NOT_FOUND).build();
     }
+    if (request.getType().isPresent() && !jobService.isValidType(request.getType().get())) {
+      return Response.status(HttpStatus.BAD_REQUEST_400)
+          .entity("Bad value for Job type " + request.getType().get())
```

Comment:
seems like this `JobType` should be a common model which could do this validation for you instead of requiring you to call into the job service for validation. Don't have to fix it here, but please address #19 afterwards.

Reply:
Yea that's a good idea. I felt a little weird about the service doing validation for a specific type - this sounds better. Are you sure #19 is the issue though?